projects/igniteui-angular/src/lib/combo/combo.pipes.ts

```diff
@@ -70,18 +70,45 @@ export class IgxComboGroupingPipe implements PipeTransform {
     }
 }
 
-function defaultFilterFunction (collection: any[], searchValue: any, filteringOptions: IComboFilteringOptions): any[] {
+function defaultFilterFunction<T>(collection: T[], searchValue: string, filteringOptions: IComboFilteringOptions): T[] {
     if (!searchValue) {
         return collection;
     }
-    const searchTerm = filteringOptions.caseSensitive ? searchValue : searchValue.toLowerCase();
-    if (filteringOptions.filteringKey != null) {
-        return collection.filter(e => filteringOptions.caseSensitive ?
-            e[filteringOptions.filteringKey]?.includes(searchTerm) :
-            e[filteringOptions.filteringKey]?.toString().toLowerCase().includes(searchTerm));
-    } else {
-        return collection.filter(e => filteringOptions.caseSensitive ?
-            e?.includes(searchTerm) :
-            e?.toString().toLowerCase().includes(searchTerm));
+
+    const { caseSensitive, filteringKey } = filteringOptions;
+    const term = caseSensitive ? searchValue : searchValue.toLowerCase();
+
+    return collection.filter(item => {
+        const str = filteringKey ? `${item[filteringKey]}` : `${item}`;
+        return (caseSensitive ? str : str.toLowerCase()).includes(term);
+    });
+}
+
+function normalizeString(str: string, caseSensitive = false): string {
+    return (caseSensitive ? str : str.toLocaleLowerCase())
+        .normalize('NFKD')
+        .replace(/\p{M}/gu, '');
+}
+
+/**
+ * Combo filter function which does not distinguish between accented letters and their base letters.
+ * For example, when filtering for "resume", this function will match both "resume" and "résumé".
+ *
+ * @example
+ * ```html
+ * <igx-combo [filterFunction]="comboIgnoreDiacriticFilterFunction"></igx-combo>
+ * ```
+ */
+export function comboIgnoreDiacriticFilterFunction<T>(collection: T[], searchValue: string, filteringOptions: IComboFilteringOptions): T[] {
```

Comment:
```suggestion
export function ignoreDiacriticsFilterFunction<T>(collection: T[], searchValue: string, filteringOptions: IComboFilteringOptions): T[] {
```

Since we already have a `defaultFilterFunction`, it makes sense to me to name the new filter function in a similar fashion.

Reply:
It does but that is in the context of the combo internal imports. Since this is exported at the top-level public API the import will look like this:
```typescript
import { ignoreDiacriticsFilterFunction } from 'igniteui-angular';
```
which does not really convey the information that this is related to the combo component.

If it was something akin to:
```typescript
import { ignoreDiacriticsFilterFunction } from 'igniteui-angular/combo';
```
then that would be better but AFAIK we do support these kind of imports.

Reply:
It does but that is in the context of the combo internal imports. Since this is exported at the top-level public API the import will look like this:
```typescript
import { ignoreDiacriticsFilterFunction } from 'igniteui-angular';
```
which does not really convey the information that this is related to the combo component.

If it was something akin to:
```typescript
import { ignoreDiacriticsFilterFunction } from 'igniteui-angular/combo';
```
then that would be better but AFAIK we do support these kind of imports.

Reply:
@jackofdiamond5 `defaultFilterFunction` is not public and it should not decide the name for the public method. It should be either descriptive enough, like your initial suggestion, @rkaraivanov: `comboIgnoreDiacriticsFilterFunction` (with Diacritic**s**?), or short enough like `comboDiacriticsFunction`?